data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/EthereumTypes.java

```diff
@@ -31,4 +39,19 @@ public class EthereumTypes {
           .description("`BLSSignature Hex` BLS12-381 signature for the current epoch.")
           .format("byte")
           .build();
+
+  public static <T extends SszData> ResponseContentTypeDefinition<T> sszResponseType(
+      final Function<T, SpecMilestone> milestoneSelector) {
+    return new OctetStreamResponseContentTypeDefinition<>(
+        SszData::sszSerialize,
+        value ->
+            Map.of(
+                RestApiConstants.HEADER_CONSENSUS_VERSION,
+                Version.fromMilestone(milestoneSelector.apply(value)).name(),
+                RestApiConstants.HEADER_CONTENT_DISPOSITION,
+                String.format(
+                    "filename=\"%s%s.ssz\"",
+                    value.getSchema().getName().map(name -> name = "-").orElse(""),
```

Comment:
shouldn't be `name + "-"`?

Reply:
The map adds a - to the end of the name if there is one.  Avoids having a leading - if there was no name.

Reply:
your code mismatches your description 😉

Reply:
Yeah I caught on to what you were saying eventually. :)